internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -681,7 +697,7 @@ func (s *Server) processRelevantRepositoryEvent(
 	if dbrepo.WebhookID.Valid {
 		// Check if the payload webhook ID matches the one we
 		// have stored in the DB for this repository
-		if event.GetHookID() != dbrepo.WebhookID.Int64 {
+		if event.GetHookID() != 0 && event.GetHookID() != dbrepo.WebhookID.Int64 {
```

Comment:
question: Does this fix work because the hook ID for repository events is always 0?

Reply:
Yes, that's exactly the point.
The issue was not there before because we dynamically looked up `hook_id` using `jq`, while now we're using a shared struct to deserialize the messages and this makes it always get the default value.

Simply checking for it being some value other than `0` is sufficient and cannot cause bugs because it is not a valid value, and `go-github` handles deserialization in the same way as we do here, so we should be on the safe side.